Add Kernel Extension for plotly formatter

WalternativE opened the pull request:
- Add dependencies for extension in own group
  because they are not stable yet
- Add fix for GenericChart JS which breaks the embedded
  charts in VSCode
- Add notebook for visually testing extension

I'd be happy to chat about how it should be incorporated best into the standing project. This is a best effort PR right now so I'd imagine, that you want to have some things changed. I'd be happy to work with you on making this right for the project 😊

WalternativE committed:
Add Kernel Extension for plotly formatter

- Add dependencies for extension in own group
  because they are not stable yet
- Add fix for GenericChart JS which breaks the embedded
  charts in VSCode
- Add notebook for visually testing extension

kMutagene commented:
Thank you for your work @WalternativE , i will check out the notebook tomorrow and play around a little before we chat about this. On first sight the only thing that i would maybe change is the namespace, 4 levels seems kinda deep

WalternativE commented:
> Thank you for your work @WalternativE , i will check out the notebook tomorrow and play around a little before we chat about this. On first sight the only thing that i would maybe change is the namespace, 4 levels seems kinda deep

Cool, looking forward to receiving your feedback 😊 Yeah, the namespace could be shorter. I just want to make sure, that it doesn't pollute other possible `Plotly.NET` namespaces. Do you have something in mind?

kMutagene commented on src/Plotly.NET.Interactive.Extension/paket.template:
```diff
@@ -0,0 +1,33 @@
+owners
+    Timo Mühlhaus, Kevin Schneider
+authors
+    Timo Mühlhaus, Kevin Schneider, F# open source contributors
```
Please take credit here ;)

kMutagene commented on Plotly.NET.sln:
```diff
@@ -83,6 +83,10 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".ci", ".ci", "{2461AFBF-6E1
 		.circleci\config.yml = .circleci\config.yml
 	EndProjectSection
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{FDA42E0F-C03B-47C1-B24B-EAEEFE281514}"
```
Very minor thing: You can add a project to the solution root with dotnet sln <sln file> add --in-root <proj file> instead of also adding the `src` folder

kMutagene commented on src/Plotly.NET.Interactive.Extension/Extension.fs:
```diff
@@ -0,0 +1,29 @@
+﻿namespace Plotly.NET.Interactive.Extension
```
I think using the namespace Plotly.NET.Interactive (dropping the '.Extension') is perfectly fine.

kMutagene commented on src/Plotly.NET/GenericChart.fs:
```diff
@@ -60,7 +60,7 @@ module HTML =
         newScript.AppendLine(@"
             var renderPlotly = function() {
-            var fsharpPlotlyRequire = requirejs.config({context:'fsharp-plotly',paths:{plotly:'https://cdn.plot.ly/plotly-latest.min'}});
+            var fsharpPlotlyRequire = requirejs.config({context:'fsharp-plotly',paths:{plotly:'https://cdn.plot.ly/plotly-latest.min'}}) || require;
```
I also saw this one in your blog post, could you give me some info why we need to do this? My js is 'basic' :wink:

WalternativE replied:
My guess would be, that hoisting works a bit different in Electron from how it works in most browsers. When calling the immediately executed anonymous function to set up the plotly chart it tries to use the `fsharpPlotlyRequire` function and dies because it is undefined at the time. In other browser environments this doesn't happen. Why does this difference exist: beats me. You can try out how it behaves in VSCode (with the .NET interactive plugin) when you don't have this expression. It should work in your normal Jupyter environment, though. I can't talk about Nteract because I didn't test it there. Should also just happen for the recent versions of VSCode Insiders because they changed something there (my guess would be, that they switched to a newer Electron version, but I'm not sure).

kMutagene replied:
Well that's strange. I guess as long as this change does not affect the other environments we can take it.

kMutagene commented on src/Plotly.NET.Interactive.Extension/Extension.fs:
```diff
@@ -0,0 +1,29 @@
+open Microsoft.DotNet.Interactive.Formatting
+open Plotly.NET.GenericChart
+
+type PlotlyNetFormatterKernelExtension() =
```
I think this can also be named just `FormatterKernelExtension` as the full name is then`Plotly.NET.Interactive.FormatterKernelExtension`

kMutagene commented on src/Plotly.NET.Interactive.Extension/Plotly.NET.Interactive.Extension.fsproj:
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
As this is a library, can we target netstandard?

WalternativE replied:
You are totally right! For some reason I thought, that the .NET Interactive nuget dependency was still using `netcoreapp` buth either they switched to `netstandard2.1` or I just mixed it up (very likely 🙈). I'll change it to `netstandard2.1`.

kMutagene commented on src/Plotly.NET.Interactive.Extension/paket.template:
```diff
@@ -0,0 +1,33 @@
+type project
+id Plotly.NET.Interactive.Extension
+title
+    Plotly.NET
```
Plotly.NET.Interactive

kMutagene reviewed (changes requested):
Awesome! The notebook just works! I find it very interesting that you are actually building the library inside the notebook. I have only highlighted a few namespace changes that would simplify things (`Plotly.NET.Interactive`). Also, please take credit in the package metadata :wink: .

WalternativE committed:
Add requested changes from PR review

- Shorten project name and related namespaces
- Reflect changes in visual inspection notebook
- Switch project to netstandard2.1

WalternativE commented:
> Awesome! The notebook just works! I find it very interesting that you are actually building the library inside the notebook. I have only highlighted a few namespace changes that would simplify things (`Plotly.NET.Interactive`). Also, please take credit in the package metadata 😉 .

Yeah, it's super neat to have a way to directly test everything from the notebook. That's how they do it in the official repo. I hope, that one day it will get easier to use relative paths in .NET Interactive.

I added all the changes you requested and retested the project. Hope I didn't forget anything. If there is something else missing, just let me know.
